rsatoolbox/vis/scatter_plot.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, Optional
+import math
+import matplotlib.pyplot
+import sklearn.manifold
+import numpy
+from rsatoolbox.util.vis_utils import weight_to_matrices, Weighted_MDS
+from rsatoolbox.vis.icon import Icon
+if TYPE_CHECKING:
+    from rsatoolbox.rdm import RDMs
+    from numpy.typing import NDArray
+    from matplotlib.figure import Figure
+seed = numpy.random.RandomState(seed=1)
+
+
+def show_scatter(
+        rdms: RDMs,
+        coords: NDArray,
+        rdm_descriptor: Optional[str]=None,
+        pattern_descriptor: Optional[str]=None,
+    ) -> Figure:
+    """Draw a 2-dimensional scatter plot based on the provided coordinates
+
+    Args:
+        rdms (RDMs): The RDMs object to display
+        coords (NDArray): Array of x and y coordinates for each
+            pattern (patterns x 2)
+        rdm_descriptor: (Optional[str]): If provided, this will be used as
+            title for each individual RDM.
+        pattern_descriptor (Optional[str]): If provided, the chosen pattern
+            descriptor will be printed adjacent to each point in the plot
+
+    Returns:
+        Figure: A matplotlib figure in which the plot is drawn
+    """
+    frac, n = math.modf(math.sqrt(rdms.n_rdm))
+    nrows, ncols = math.floor(n), math.floor(n)
```

Comment:
Wonder if we should have a consistent utility function that handles this subplot stuff, and allows setting either nrow or ncol, see

https://github.com/rsagroup/rsatoolbox/blob/28aeedc6855046f08bf178ca06b3ee36d647b3aa/rsatoolbox/vis/rdm_plot.py#L111-L120

Reply:
(also mplstyle stuff - might help achieve a consistent look and feel across plot code)

https://github.com/rsagroup/rsatoolbox/blob/28aeedc6855046f08bf178ca06b3ee36d647b3aa/rsatoolbox/vis/rdm_plot.py#L37

Reply:
Absolutely, made this issue #207